mar/lens/command.hoon

```diff
@@ -2,13 +2,13 @@
 /+    old-zuse
 ::
 =,  old-zuse
-|_  com/command:lens
+|_  com/command:^^lens
```

Comment:
@belisarius222, I found this by trial and error. I can't make heads or tails of it. Any idea what's happening here?

Reply:
Could be a result of changes to how =, works. Me and Ted noticed we had to do `..^is` in a jet hint because there was an additional copy included by =,

Reply:
Looks like old-zuse defines `lens`, and then we're skipping over the imported sur/lens, to grab `lens` from the standard library? I'm still reading through this.

Reply:
There's no `+lens` in zuse or `old-zuse`. There's a `+lens` in hoon.hoon, but it doesn't have a `+command` in it.

I suspect `=,  old-zuse` exposes two copies of the standard library, each of which has the `lens` from hoon.hoon.

A downside of the way organic dairy imports work is that they can expose multiple copies of the standard library, especially if you `=,` them. Arguably, Ford should try to convert imported cores to lead (opaque) cores, although that could break other things, and Hoon files aren't constrained to producing cores, so that functionality would break if a library produces, say, a cell of an atom and a core. Another alternative would be making `=,` of a core convert the core to a lead core, although that would suffer from the same issue. The last option I can see would be a convention to do `=,  ^?(lens)`.

In my own code I generally only do `=,`s inside functions, since each one raises the risk of some nasty namespace issue. Also, when I see a symbol, I like to see where it came from. This is a bit of a cop-out, though.

Reply:
@eglaysher you might have some ideas here

Reply:
Oh, there is yet another option, which would be to have a convention to `^?` cores produced by libraries if they're not supposed to expose the standard library. The system would need to be resilient against the inevitable lack of compliance with this convention, though.

Reply:
Thanks for checking. It's worth noting that `command:lens` doesn't work in this case, and neither does `command:^lens`.  IMO, old-zuse should really be removed on release-candidate. That would certainly make this sort of thing less likely. Do we need a separate issue to track namespace oddities?

Reply:
Maybe we should add a `^?` to `old-zuse`. Wanna give that a shot?

Reply:
I'm not opposed to having a catch-all issue for dealing with namespace weirdnesses, if you think that's a good way to go.

Reply:
Removing old-zuse entirely is not a bad idea. It's tedious, but it would be a good task for someone who's just learning Hoon.